src/core/permissions/sagas/updatePermissions.js

```diff
@@ -32,32 +38,47 @@ function* updatePermissionsWorker(action :SequenceAction) :Saga<WorkerResponse>
   try {
     yield put(updatePermissions.request(action.id, action.value));
 
-    const {
+    const permissionsUpdates :Array<{
+      actionType :ActionType;
+      permissions :Map<List<UUID>, Ace>;
+    }> = action.value;
+
+    const updateAclsCalls = [];
+
+    permissionsUpdates.forEach(({
       actionType,
       permissions,
     } :{
       actionType :ActionType;
       permissions :Map<List<UUID>, Ace>;
-    } = action.value;
+    }) => {
+
+      const updates = [];
+
+      permissions.forEach((ace :Ace, key :List<UUID>) => {
 
-    const updates = [];
-    permissions.forEach((ace :Ace, key :List<UUID>) => {
+        const acl = (new AclBuilder())
+          .setAces([ace])
+          .setAclKey(key)
+          .build();
 
-      const acl = (new AclBuilder())
-        .setAces([ace])
-        .setAclKey(key)
-        .build();
+        const aclData = (new AclDataBuilder())
+          .setAcl(acl)
+          .setAction(actionType)
+          .build();
 
-      const aclData = (new AclDataBuilder())
-        .setAcl(acl)
-        .setAction(actionType)
-        .build();
+        updates.push(aclData);
+      });
 
-      updates.push(aclData);
+      updateAclsCalls.push(call(updateAclsWorker, updateAcls(updates)));
```

Comment:
You don't need to do this. You don't need one `call` per item in the array because `updateAcls` is the bulk endpoint. All you need is a single array of `aclData` objects:

```
const updates = [];
permissionsUpdates.forEach(({ actionType, permissions }) => {
  permissions.forEach((ace :Ace, key :List<UUID>) => {
    ...
    updates.push(aclData);
  });
});

const response :WorkerResponse = yield call(updateAclsWorker, updateAcls(updates));
if (response.error) throw response.error;
```

Reply:
oops, you're right. thanks!